var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -270,12 +271,22 @@ def setup_dependent_package(self, module, dependent_spec):
             mkdirp(module.perl_lib_dir)
 
     def setup_build_environment(self, env):
+        spec = self.spec
+
         # This is to avoid failures when using -mmacosx-version-min=11.1
         # since not all Apple Clang compilers support that version range
         # See https://eclecticlight.co/2020/07/21/big-sur-is-both-10-16-and-11-0-its-official/
-        if self.spec.satisfies('os=bigsur'):
+        if spec.satisfies('os=bigsur'):
             env.set('SYSTEM_VERSION_COMPAT', 1)
 
+        # This is how we tell perl the locations of bzip and zlib.
+        env.set('BUILD_BZIP2', 0)
+        env.set('BZIP2_INCLUDE', spec['bzip2'].prefix.include)
+        env.set('BZIP2_LIB',     spec['bzip2'].prefix.lib)
+        env.set('BUILD_ZLIB', 0)
+        env.set('ZLIB_INCLUDE', spec['zlib'].prefix.include)
+        env.set('ZLIB_LIB',     spec['zlib'].prefix.lib)
```

Comment:
This won't work if the libraries are installed in lib64. We should use libs.directories[0] instead.